Q1: What is the difference between a t-test and a z-test? Provide an example scenario where you would
use each type of test.

Answer:-

A t-test and a z-test are both statistical tests used to make inferences about population parameters based on sample data. However, there are some differences between the two tests, primarily related to the assumptions made about the population and the sample size.

A z-test assumes that the sample comes from a normally distributed population, and the population standard deviation is known. This test is used when the sample size is large (typically, n > 30) and the population variance is known or can be assumed to be known. For example, if a company wants to test if the mean height of their employees is different from the national average, and they know the standard deviation of the population height, they can use a z-test.

On the other hand, a t-test assumes that the sample comes from a normally distributed population, and the population standard deviation is unknown. This test is used when the sample size is small (typically, n < 30) or when the population variance is unknown. For example, if a researcher wants to test if a new drug has a different effect than a placebo, they can use a t-test to compare the mean scores of two groups of participants.

Example scenarios:

A company wants to compare the mean salary of their employees to the national average salary. They have a large sample size (n>30) and know the population standard deviation. They can use a z-test to make this comparison.

A researcher wants to compare the mean test scores of two groups of students. They have a small sample size (n<30) and do not know the population standard deviation. They can use a t-test to make this comparison.

Q2: Differentiate between one-tailed and two-tailed tests.

Answer:-

In hypothesis testing, the null hypothesis (H0) is a statement about a population parameter that is assumed to be true unless there is sufficient evidence to reject it in favor of an alternative hypothesis (Ha). The alternative hypothesis is the statement that the researcher is trying to support.

One-tailed tests and two-tailed tests are types of hypothesis tests that differ in the directionality of the alternative hypothesis.

A one-tailed test is a hypothesis test in which the alternative hypothesis specifies the direction of the effect or difference being tested. That is, the researcher is interested in whether the population parameter is either greater than or less than a certain value, but not both. Therefore, the rejection region for the null hypothesis is only on one side of the distribution. A one-tailed test is often used when the researcher has a specific directional hypothesis or when there is a theoretical or practical reason to expect an effect in a specific direction.

For example, a one-tailed test might be used to test the hypothesis that a new medication will decrease the amount of time it takes for patients to fall asleep. The alternative hypothesis would be that the medication decreases the amount of time it takes for patients to fall asleep, and the null hypothesis would be that the medication has no effect on the amount of time it takes for patients to fall asleep.

In contrast, a two-tailed test is a hypothesis test in which the alternative hypothesis does not specify the direction of the effect or difference being tested. That is, the researcher is interested in whether the population parameter is different from a certain value in any direction. Therefore, the rejection region for the null hypothesis is split between both sides of the distribution. A two-tailed test is often used when the researcher has no specific directional hypothesis or when there is no theoretical or practical reason to expect an effect in a specific direction.

For example, a two-tailed test might be used to test the hypothesis that the mean weight of male and female newborns is the same. The alternative hypothesis would be that the mean weight of male and female newborns is different, and the null hypothesis would be that the mean weight of male and female newborns is equal.

In summary, one-tailed tests and two-tailed tests differ in the directionality of the alternative hypothesis, and the choice between the two depends on the research question and the directional nature of the hypothesis.

Q3: Explain the concept of Type 1 and Type 2 errors in hypothesis testing. Provide an example scenario for
each type of error.

Answer:-

In hypothesis testing, there are two types of errors that can occur: Type I error and Type II error.

Type I error occurs when the null hypothesis is rejected even though it is actually true. This means that the researcher concludes that there is a significant effect or difference between two groups when in reality there is not. The probability of making a Type I error is denoted as alpha (α) and is set by the researcher prior to conducting the test. A smaller alpha level means a lower probability of making a Type I error.

Type II error, on the other hand, occurs when the null hypothesis is not rejected even though it is actually false. This means that the researcher concludes that there is no significant effect or difference between two groups when in reality there is. The probability of making a Type II error is denoted as beta (β). The probability of making a Type II error depends on several factors such as the sample size, effect size, and the alpha level.

Here are example scenarios for each type of error:

Type I error: Suppose a company produces a batch of 10,000 pens and claims that the average weight of each pen is 5 grams. A researcher randomly selects a sample of 50 pens and finds that the average weight is 5.2 grams with a standard deviation of 0.5 grams. The researcher conducts a hypothesis test with a significance level of 0.05 and concludes that the average weight of the pens is significantly greater than 5 grams. However, in reality, the average weight of the pens is actually 5 grams, and the researcher has made a Type I error by rejecting the null hypothesis.

Type II error: Consider a clinical trial testing the effectiveness of a new drug for treating a certain disease. The null hypothesis is that the drug has no effect on the disease, while the alternative hypothesis is that the drug does have an effect. If the clinical trial fails to reject the null hypothesis, it could be due to the drug having no effect, or it could be due to a Type II error, where the drug actually does have an effect but it was not detected in the study. In this scenario, a Type II error would occur if the new drug is effective, but the clinical trial fails to detect its effectiveness due to a small sample size or other limitations of the study.



Q4: Explain Bayes's theorem with an example.

Answer:-

Bayes's theorem is a mathematical formula used to calculate the conditional probability of an event based on prior knowledge of related events. It is named after Thomas Bayes, an 18th-century statistician, and philosopher.

Bayes's theorem can be expressed as follows:

P(A|B) = P(B|A) * P(A) / P(B)

where P(A|B) is the probability of event A given event B has occurred, P(B|A) is the probability of event B given that event A has occurred, P(A) is the prior probability of event A, and P(B) is the prior probability of event B.

An example scenario where Bayes's theorem can be applied is in medical diagnosis. Suppose a patient exhibits certain symptoms that may indicate the presence of a disease. The probability of having the disease given the symptoms can be calculated using Bayes's theorem.

Let's say that the disease is present in 1% of the population, and the probability of a person exhibiting the symptoms given that they have the disease is 90%. On the other hand, the probability of a person exhibiting the symptoms without having the disease is 10%. Using this information, we can apply Bayes's theorem to calculate the probability of a person having the disease given that they exhibit the symptoms.

P(Disease|Symptoms) = P(Symptoms|Disease) * P(Disease) / P(Symptoms)

where P(Disease|Symptoms) is the probability of having the disease given the symptoms, P(Symptoms|Disease) is the probability of exhibiting the symptoms given the disease, P(Disease) is the prior probability of having the disease, and P(Symptoms) is the prior probability of exhibiting the symptoms.

Plugging in the values, we get:

P(Disease|Symptoms) = 0.90 * 0.01 / ((0.90 * 0.01) + (0.10 * 0.99))

= 0.08

Therefore, the probability of having the disease given the symptoms is 8%. This means that even though the patient exhibits the symptoms, there is still only an 8% chance of having the disease.

Q5: What is a confidence interval? How to calculate the confidence interval, explain with an example.

Answer:-

A confidence interval is a range of values that is likely to contain the true value of a population parameter with a certain degree of confidence. In other words, it is a range of values around a sample statistic that is likely to contain the true population parameter value with a certain level of certainty or probability.

To calculate a confidence interval, we need to know the sample mean, sample size, and standard deviation (or standard error) of the sample statistic. The formula for a confidence interval is:

Confidence interval = sample mean ± (z-score * standard error)

where z-score is the number of standard deviations corresponding to the desired level of confidence. For example, for a 95% confidence interval, the z-score would be 1.96.

Here's an example of how to calculate a confidence interval:

Suppose we want to estimate the average height of all people in a city. We take a random sample of 50 people and find that the sample mean height is 170 cm, and the sample standard deviation is 5 cm.

To calculate a 95% confidence interval for the population mean height, we need to first calculate the standard error. The standard error is calculated as:

Standard error = sample standard deviation / sqrt(sample size)

= 5 / sqrt(50)

= 0.71

Next, we can calculate the confidence interval as:

Confidence interval = 170 ± (1.96 * 0.71)

= 168.6 to 171.4 cm

This means that we can be 95% confident that the true population mean height falls within the range of 168.6 to 171.4 cm. We can interpret this as if we repeat this sampling process many times, 95% of the time, the true population mean height will be contained within this interval.

Q6. Use Bayes' Theorem to calculate the probability of an event occurring given prior knowledge of the
event's probability and new evidence. Provide a sample problem and solution.

Answer:-

here's an example problem that uses Bayes' theorem:

Suppose a new medical test has been developed to detect a certain disease, which affects 1% of the population. The test has a false positive rate of 5%, meaning that 5% of the time, a healthy person will test positive for the disease. The test also has a false negative rate of 10%, meaning that 10% of the time, a person with the disease will test negative. Suppose a person takes the test and receives a positive result. What is the probability that the person actually has the disease?

Let's define some terms first:

A: the event that the person has the disease B: the event that the person tests positive for the disease We want to find P(A|B), the probability that the person has the disease given a positive test result. We can use Bayes' theorem to calculate this as follows:

P(A|B) = P(B|A) * P(A) / P(B)

where P(B|A) is the probability of testing positive given that the person has the disease, P(A) is the prior probability of the person having the disease, and P(B) is the probability of testing positive for the disease.

Using the information given in the problem, we can calculate these probabilities as follows:

P(B|A) = 1 - 0.10 = 0.90 (the probability of testing positive given that the person has the disease is 90%) P(A) = 0.01 (the prior probability of the person having the disease is 1%) P(B) = P(B|A) * P(A) + P(B|not A) * P(not A) = 0.90 * 0.01 + 0.05 * 0.99 (the probability of testing positive is the sum of the probability of a true positive and a false positive) Plugging these values into the Bayes' theorem formula, we get:

P(A|B) = 0.90 * 0.01 / (0.90 * 0.01 + 0.05 * 0.99)

= 0.15

Therefore, the probability that the person actually has the disease given a positive test result is 15%. This means that the test is not very accurate, and there is still a high chance of a false positive result.

Q7. Calculate the 95% confidence interval for a sample of data with a mean of 50 and a standard deviation
of 5. Interpret the results.

Answer:-

Sample Size is not provided in above Question , I am assuming sample size of 60

In [3]:
import math
import scipy.stats as stats

# Define the sample mean, standard deviation, and sample size
sample_mean = 50
sample_std_dev = 5
sample_size = 60

# Signficance Level
alpha = 0.05

# Calculate the critical value for a 95% confidence interval
t_crit = stats.t.ppf(1-alpha/2, df=sample_size-1)
print(f't statistic for significance level {alpha} , sample size of {sample_size} is : {t_crit:.2f}')

# Calculate the standard error
std_error = sample_std_dev / math.sqrt(sample_size)

# Calculate the lower and upper bounds of the confidence interval
lower_bound = sample_mean - t_crit * std_error
upper_bound = sample_mean + t_crit * std_error

# Print the results
print(f"The 95% confidence interval is ({lower_bound:.2f}, {upper_bound:.2f})")

t statistic for significance level 0.05 , sample size of 60 is : 2.00
The 95% confidence interval is (48.71, 51.29)


Q8. What is the margin of error in a confidence interval? How does sample size affect the margin of error?
Provide an example of a scenario where a larger sample size would result in a smaller margin of error.

Answer:-

The margin of error in a confidence interval is the range of values around the sample estimate that we are confident contains the true population value. It represents the amount of uncertainty in our estimate due to sampling error.

The margin of error is affected by several factors, including the level of confidence desired, the sample size, and the variability of the data. In general, a larger sample size results in a smaller margin of error, all else being equal. This is because a larger sample size reduces the effect of random sampling error, and thus increases the precision of the estimate.

For example, suppose we want to estimate the proportion of adults in a city who support a particular political candidate, and we take a random sample of 100 adults. The sample proportion is 0.6, which means that 60% of the sample supports the candidate. We want to construct a 95% confidence interval for the true population proportion.

Assuming a normal distribution and using the formula for the margin of error, we can calculate:

Margin of error = z * sqrt(p_hat * (1 - p_hat) / n)

where z is the critical value for a 95% confidence interval, p_hat is the sample proportion, and n is the sample size.

Using a standard normal table, we find that the critical value for a 95% confidence interval is 1.96. Plugging in the values, we get:

Margin of error = 1.96 * sqrt(0.6 * 0.4 / 100)

= 0.098

This means that we are 95% confident that the true population proportion of adults who support the candidate is between 0.502 and 0.698 (0.6 ± 0.098).

Now, suppose we increase the sample size to 400. Using the same formula, we can calculate the new margin of error as:

Margin of error = 1.96 * sqrt(0.6 * 0.4 / 400)

= 0.049

This means that we are 95% confident that the true population proportion of adults who support the candidate is between 0.551 and 0.649 (0.6 ± 0.049).

As we can see, the larger sample size results in a smaller margin of error, which means that our estimate is more precise and we can be more confident in our conclusion.



Q9. Calculate the z-score for a data point with a value of 75, a population mean of 70, and a population
standard deviation of 5. Interpret the results.

Answer:-

Below is my python code to calculate above Z Score

In [5]:
# Given data
data_point = 75
population_mean = 70
population_std_dev = 5

# Calculate the z-score
z_score = (data_point - population_mean) / population_std_dev

# Print the result
print(f"The z-score is: {z_score}")

The z-score is: 1.0


This means that the data point is 1 standard deviation above the population mean. A positive z-score indicates that the data point is above the mean, while a negative z-score indicates that it is below the mean. The magnitude of the z-score tells us how far away the data point is from the mean in terms of standard deviations.

In [6]:
from scipy.stats import norm

# Calculate the percentage of values below a z-score of 1
percent_below = norm.cdf(z) * 100

print(f"The percentage of values below a z-score of 1 is: {percent_below:.2f}%")

The percentage of values below a z-score of 1 is: 84.13%


A z-score of 1 corresponds to approximately 84.13% of values being below this value in a standard normal distribution.

Q10. In a study of the effectiveness of a new weight loss drug, a sample of 50 participants lost an average
of 6 pounds with a standard deviation of 2.5 pounds. Conduct a hypothesis test to determine if the drug is
significantly effective at a 95% confidence level using a t-test.

Answer:-



In [7]:
import numpy as np
from scipy.stats import t

# Given data
sample_mean = 6  # average weight loss
sample_std_dev = 2.5  # standard deviation of weight loss
sample_size = 50  # number of participants
population_mean = 0  # mean weight loss under null hypothesis

# Calculate the t-statistic
t_statistic = (sample_mean - population_mean) / (sample_std_dev / (sample_size ** 0.5))

# Calculate the degrees of freedom
df = sample_size - 1

# Calculate the critical t-value for a one-tailed test at 95% confidence level
alpha = 0.05
critical_t_value = stats.t.ppf(1 - alpha, df)

# Calculate the p-value
p_value = 1 - stats.t.cdf(t_statistic, df)

# Print the results
print(f"t-Statistic: {t_statistic:.2f}")
print(f"Critical t-Value: {critical_t_value:.2f}")
print(f"P-Value: {p_value:.4f}")

# Decision rule
if t_statistic > critical_t_value:
    print("Reject the null hypothesis: The drug is significantly effective.")
else:
    print("Fail to reject the null hypothesis: The drug is not significantly effective.")


t-Statistic: 16.97
Critical t-Value: 1.68
P-Value: 0.0000
Reject the null hypothesis: The drug is significantly effective.


Q11. In a survey of 500 people, 65% reported being satisfied with their current job. Calculate the 95%
confidence interval for the true proportion of people who are satisfied with their job.

Answer:-


In [8]:
import math

p = 0.65  # sample proportion
n = 500  # sample size
z_alpha = 1.96  # z-score for 95% confidence level

# Calculate the standard error
se = math.sqrt((p * (1 - p)) / n)

# Calculate the margin of error
me = z_alpha * se

# Calculate the confidence interval
lower_bound = p - me
upper_bound = p + me

# Print the results
print(f"The 95% confidence interval for the proportion of people who are satisfied with their job is ({lower_bound*100:.2f}%, {upper_bound*100:.2f}%)")


The 95% confidence interval for the proportion of people who are satisfied with their job is (60.82%, 69.18%)


Q12. A researcher is testing the effectiveness of two different teaching methods on student performance.
Sample A has a mean score of 85 with a standard deviation of 6, while sample B has a mean score of 82
with a standard deviation of 5. Conduct a hypothesis test to determine if the two teaching methods have a
significant difference in student performance using a t-test with a significance level of 0.01.

Answer:-

In [9]:
import math
from scipy import stats

# Given data for Sample A
mean_A = 85
std_dev_A = 6
n_A = 30

# Given data for Sample B
mean_B = 82
std_dev_B = 5
n_B = 30

# Calculate the t-statistic
pooled_std_error = math.sqrt((std_dev_A**2 / n_A) + (std_dev_B**2 / n_B))
t_statistic = (mean_A - mean_B) / pooled_std_error

# Calculate the degrees of freedom
numerator = ((std_dev_A**2 / n_A) + (std_dev_B**2 / n_B))**2
denominator = ((std_dev_A**2 / n_A)**2 / (n_A - 1)) + ((std_dev_B**2 / n_B)**2 / (n_B - 1))
degrees_of_freedom = numerator / denominator

# Calculate the critical t-value for a two-tailed test at 0.01 significance level
alpha = 0.01
critical_t_value = stats.t.ppf(1 - alpha/2, degrees_of_freedom)

# Calculate the p-value
p_value = 2 * (1 - stats.t.cdf(abs(t_statistic), degrees_of_freedom))

# Print the results
print(f"t-Statistic: {t_statistic:.4f}")
print(f"Degrees of Freedom: {degrees_of_freedom:.4f}")
print(f"Critical t-Value: {critical_t_value:.4f}")
print(f"P-Value: {p_value:.4f}")

# Decision rule
if p_value < alpha:
    print("Reject the null hypothesis: The two teaching methods have a significant difference in student performance.")
else:
    print("Fail to reject the null hypothesis: The two teaching methods do not have a significant difference in student performance.")


t-Statistic: 2.1039
Degrees of Freedom: 56.1733
Critical t-Value: 2.6662
P-Value: 0.0399
Fail to reject the null hypothesis: The two teaching methods do not have a significant difference in student performance.


Q13. A population has a mean of 60 and a standard deviation of 8. A sample of 50 observations has a mean
of 65. Calculate the 90% confidence interval for the true population mean.

Answer:-


In [10]:
import math
from scipy.stats import t

pop_mean = 60
pop_std = 8
n = 50
sample_mean = 65
confidence_level = 0.90

# Calculate the t-score
t_score = (sample_mean - pop_mean) / (pop_std / math.sqrt(n))

# Find the critical t-value
df = n - 1
t_crit = t.ppf((1+confidence_level)/2, df)
print(f't statistic for {n} samples with {confidence_level*100:.0f}% confidence is : {t_crit:.4f}')

# Calculate the margin of error
margin_of_error = t_crit * (pop_std / math.sqrt(n))

# Calculate the confidence interval
lower_bound = sample_mean - margin_of_error
upper_bound = sample_mean + margin_of_error

# Print the results
print(f"The {confidence_level*100:.0f}% confidence interval for the true population mean is ({lower_bound:.3f}, {upper_bound:.3f})")


t statistic for 50 samples with 90% confidence is : 1.6766
The 90% confidence interval for the true population mean is (63.103, 66.897)


Q14. In a study of the effects of caffeine on reaction time, a sample of 30 participants had an average
reaction time of 0.25 seconds with a standard deviation of 0.05 seconds. Conduct a hypothesis test to
determine if the caffeine has a significant effect on reaction time at a 90% confidence level using a t-test.

Answer:-

In [11]:
import math
from scipy import stats

# Given data
sample_mean = 0.25
sample_std_dev = 0.05
sample_size = 30
baseline_mean = 0.30
confidence_level = 0.90

# Calculate the t-statistic
t_statistic = (sample_mean - baseline_mean) / (sample_std_dev / math.sqrt(sample_size))

# Calculate the degrees of freedom
df = sample_size - 1

# Calculate the critical t-value for a two-tailed test at 90% confidence level
alpha = 1 - confidence_level
critical_t_value = stats.t.ppf(1 - alpha / 2, df)

# Calculate the p-value
p_value = 2 * (1 - stats.t.cdf(abs(t_statistic), df))

# Print the results
print(f"t-Statistic: {t_statistic:.4f}")
print(f"Degrees of Freedom: {df}")
print(f"Critical t-Value: {critical_t_value:.4f}")
print(f"P-Value: {p_value:.4f}")

# Decision rule
if abs(t_statistic) > critical_t_value:
    print("Reject the null hypothesis: Caffeine has a significant effect on reaction time.")
else:
    print("Fail to reject the null hypothesis: Caffeine does not have a significant effect on reaction time.")

t-Statistic: -5.4772
Degrees of Freedom: 29
Critical t-Value: 1.6991
P-Value: 0.0000
Reject the null hypothesis: Caffeine has a significant effect on reaction time.
